qoa.h

```diff
@@ -391,8 +392,14 @@ unsigned int qoa_encode_frame(const short *sample_data, qoa_desc *qoa, unsigned
 			qoa_uint64_t best_error = -1;
 			qoa_uint64_t best_slice;
 			qoa_lms_t best_lms;
+			int cur_best_scalefactor = 0;
 
-			for (int scalefactor = 0; scalefactor < 16; scalefactor++) {
+			for (int _scalefactor = 0; _scalefactor < 16; _scalefactor++) {
```

Comment:
`_scalefactor` is quite a weird name. Maybe `sfi` (for scalefactor index/iterator), analogous to `si` (sample index/iterator) elsewhere?

Reply:
Hah, I literally had `sfi` before changing it to `_scalefactor`. Will change it back.

Maybe keep the `cur_best_scalefactor` analogous to `best_error`, `best_slice`, etc as well? So, `best_scalefactor` for the current slice, and an array `prev_best_scalefactors` or `scalefactor_hist` outside the loop?

Reply:
> Maybe keep the `cur_best_scalefactor` analogous to `best_error`, `best_slice`, etc as well? So, `best_scalefactor` for the current slice

Sounds good!

> and an array `prev_best_scalefactors` or `scalefactor_hist` outside the loop?

I propose just `prev_scalefactor` for the array. But either of your suggestions are good, too :)